### DISCUSSION TITLE
UART over micropython PIO

### DISCUSSION THREAD

**Discussion by mathur-exe** [micropython/micropython#14246] [RP2040 / Pico]
I've been trying to achieve UART communication between two RPi Pico using Micropython PIO. But haven't been able to receive data 

My Tx & Rx Pins are 0 and 1

These are the codes I'm following
- [micropython/pio_uart_rx.py](https://github.com/micropython/micropython/blob/master/examples/rp2/pio_uart_rx.py)
- [micropython/pio_uart_tx.py](https://github.com/micropython/micropython/blob/master/examples/rp2/pio_uart_tx.py)

This is the output I'm getting for the reception
```
>>> %Run -c $EDITOR_CONTENT
Hello, world from PIO, using uart_rx_mini!
Hello, world from PIO, using uart_rx!
```

**Comment by Rumidom** [comment]
have you swapped Rx and Tx pins?  Rx of the first Pi connects to TX on the second pi, and Tx on the first Pi connects to RX on the second Pi

**Comment by mathur-exe** [comment]
@Rumidom 
Yes, i've swapped the Tx and Rx of the Pico

Context: I am trying to achieve low latency (micro or nanoseconds) communication between pico. I've tried using micropython's UART but it has a very high latency. But the reception was working. In case of micropython PIO, I'm able to achieve transmission at nanoseconds latency at 125MHz but I'm not able to receive it
